### ISSUE TITLE
With Intel ifort, example/tests is stopped before the end

### ISSUE THREAD

**Issue by vmagnin** [vmagnin/gtk-fortran#238]
**Describe the bug**
When gtk-fortran is built with ifort, example/tests is stopping before it is finished:
```bash
$ ./tests
 test_iso_c_binding()
 test_c_char_in_out()
 test_guchar_in_out()
 test_gdouble_in_out()
 test_gulong_in()
 test_int16_in_out()
 test_uint16_in_out()
 test_int32_in_out()
Process stopped
```

**Expected behavior**
Should continue and finally print:
```
 test_uint32_in_out()
 test_gboolean_in_out()
           1           1
           0           0

 No error
```

**Your system:**
 - OS version: Ubuntu 21.04 64 bits
 - Compiler version: ifort (IFORT) 2021.2.0 20210228
 - GTK branch: tested only in gtk4.

**Additional context**
No problem with gfortran.

**Comment by vmagnin** [comment]
With GTK 4.4, GLib 2.70.0 and ifort 2021.4.0 under Ubuntu 21.10, the `tests` application is quickly filling the whole RAM. If it stopped by CTRL+C, the following messages are shown:
```bash
 test_int32_in_out()
^Cforrtl: error (69): process interrupted (SIGINT)
Image              PC                Routine            Line        Source             
tests              000000000040635B  Unknown               Unknown  Unknown
libc.so.6          00007F1A50333520  Unknown               Unknown  Unknown
libc.so.6          00007F1A50394AE6  Unknown               Unknown  Unknown
libc.so.6          00007F1A50395B2F  malloc                Unknown  Unknown
libglib-2.0.so.0.  00007F1A50657089  g_malloc              Unknown  Unknown
libglib-2.0.so.0.  00007F1A5066BEA6  g_memdup2             Unknown  Unknown
libglib-2.0.so.0.  00007F1A50624128  g_bytes_new           Unknown  Unknown
libglib-2.0.so.0.  00007F1A5068A7DE  g_variant_new_int     Unknown  Unknown
tests              00000000004044BA  Unknown               Unknown  Unknown
tests              0000000000403962  Unknown               Unknown  Unknown
libc.so.6          00007F1A5031AFD0  Unknown               Unknown  Unknown
libc.so.6          00007F1A5031B07D  __libc_start_main     Unknown  Unknown
tests              0000000000403865  Unknown               Unknown  Unknown
```
Idea: try to call `g_variant_unref()`: https://developer-old.gnome.org/glib/unstable/glib-GVariant.html#g-variant-unref

**Comment by vmagnin** [comment]
Freeing each g_variant with `g_variant_unref()` avoid RAM filling. But the program still does not stop. This is due to that loop:
```fortran
    do a = -huge(b), huge(b), +65536
```
That loop is undefined as it can overflow. With some compilers, it will run OK (GFortran), with others a will become <0 and the loop will never end (ifort).

Fixed by writing:
```fortran
    do a = -huge(b), huge(b)-65536, +65536
```
Fixed in the gtk4 branch (will be included in gtk-fortran 4.2).
